src/notebook/widget.ts

```diff
@@ -391,36 +444,79 @@ class NotebookWidget extends Widget {
     let layout = this._notebook.layout as PanelLayout;
     let constructor = this.constructor as typeof NotebookWidget;
     let factory = constructor.createCell;
-    let widget: Widget;
+    let widget: BaseCellWidget;
     switch(args.type) {
     case ListChangeType.Add:
-      layout.insertChild(args.newIndex, factory(args.newValue as ICellModel));
+      widget = factory(args.newValue as ICellModel);
+      layout.insertChild(args.newIndex, widget);
+      this.initCell(widget);
```

Comment:
This case can be reverted to the old code when we just add a single focus/blur handler to the notebook widget.